File: java/catraca_virtual/catraca_virtual/src/main/java/br/edu/unilab/unicafe/model/Cartao.java
package br.edu.unilab.unicafe.model;

public class Cartao {
	
	private int id;
	private String numero;
	private double creditos;
	private Tipo tipo;
	
	public Cartao(){
		this.tipo = new Tipo();
	}
	
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	
	public String getNumero() {
		return numero;
	}
	public void setNumero(String numero) {
		this.numero = numero;
	}
	public double getCreditos() {
		return creditos;
	}
	public void setCreditos(double creditos) {
		this.creditos = creditos;
	}
	public Tipo getTipo() {
		return tipo;
	}
	public void setTipo(Tipo tipo) {
		this.tipo = tipo;
	}
}

File: classes/model/Cartao.php
<?php

class Cartao{
	private $id;
	private $numero;
	private $creditos;
	private $tipo;
	
	
	
	public function __construct(){
		$this->tipo = new Tipo();
		$this->creditos = 0;
		$this->id = 0;
	}
	
	public function setId($id){
		$this->id = $id;
		
	}
	public function getId(){
		return $this->id;
	}
	public function setNumero($numero){
		$this->numero = preg_replace ('/[^0-9]/', '', $numero);
	}
	public function getNumero(){
		return $this->numero;
	}
	public function setCreditos($creditos){
		$this->creditos = $creditos;
	}
	public function getCreditos(){
		return $this->creditos;
	}
	public function setTipo(Tipo $tipo){
		$this->tipo = $tipo;
	}
	public function getTipo(){
		return $this->tipo;
	}
	public function adicionaCreditos($creditos){
		if(is_numeric($creditos)){
			$this->setCreditos($this->getCreditos()+$creditos);
		}
	}
}





?>